src/_pytest/_io/__init__.py

```diff
@@ -19,4 +23,44 @@ def _highlight(self, source):
         except ImportError:
             return source
         else:
-            return highlight(source, PythonLexer(), TerminalFormatter())
+            return highlight(source, PythonLexer(), TerminalFormatter(bg="dark"))
+
+
+@attr.s(frozen=True)
+class ParsedTraceback:
+    sources = attr.ib()  # type: Tuple[str, ...]
+    errors = attr.ib()  # type: Tuple[str, ...]
+    explanations = attr.ib()  # type: Tuple[str, ...]
+
+    ERROR_PREFIX = ">    "
+    EXPLANATION_PREFIX = "E    "
```

Comment:
Instead of parsing it out of Repr's again (`ReprEntry` here) it might make sense to extend those instead?

Reply:
You mean adding this function/utility to `ReprEntry`? Or do you mean something else?

Another thing your comment made me consider was to "follow the source" (no pun intended), in which case I can see that the markers are generated in `FormattedExcinfo.get_source`. 

Ideally I would like for it to return actually structured data instead of formatted lines, so we can re-use it later without re-parsing, unfortunately this might be considered a breaking change. Will think about it some more (meanwhile suggestions are welcome).

Reply:
Yes, structured data would be much better.
This would also help when it comes to highlighting different parts differently (e.g. for https://github.com/pytest-dev/pytest/issues/522).